### PR TITLE
fix: Add viewAppearing to UIViewController spans

### LINKED ISSUE #1198: Add span viewAppearing
The auto UI performance instrumentation generates transactions with missing instrumentation, see the screenshots below.

image
image

Looking at the state transition from [`UIViewController`](https://developer.apple.com/documentation/uikit/uiviewcontroller) it makes sense to add an extra span called `viewAppearing` to fill this gap.

image

The span starts after measuring the performance of [`viewWillAppear`](https://github.com/getsentry/sentry-cocoa/blob/dd9501e8903260019f5b0fe05e73f25093201df2/Sources/Sentry/SentryUIViewControllerPerformanceTracker.m#L69) and finishes before measuring the performance of [`viewDidAppear`](https://github.com/getsentry/sentry-cocoa/blob/dd9501e8903260019f5b0fe05e73f25093201df2/Sources/Sentry/SentryUIViewControllerPerformanceTracker.m#L74).

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Included viewAppearing span in ui performance tracker to measure time that takes a UIViewController to appear.

## :bulb: Motivation and Context

Closes #1198 

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

_#skip-changelog_

### CODE REVIEW

**Comment by philipphofmann** [comment]
I think there is something not working properly. Please take a look at [this transaction](https://sentry.io/organizations/sentry-sdks/performance/sentry-cocoa:9ce255af1ff349f9a094feb168d1f5c3/?project=5428557&query=transaction.duration%3A%3C15m+event.type%3Atransaction&showTransactions=recent&statsPeriod=24h&transaction=iOS_Swift.ViewController&unselectedSeries=p100%28%29)

I see a few spans multiple times.

Screen Shot 2021-07-02 at 11 32 03

**Comment by philipphofmann** [comment]
Screen Shot 2021-07-05 at 16 54 21


@brustolin does it maybe make sense to make the subview spans, that I marked in an orange rectangle, children of `viewAppearing`, as they are always executed during `viewAppearing`?

**Comment by brustolin** [comment]
> Screen Shot 2021-07-05 at 16 54 21
> 
> @brustolin does it maybe make sense to make the subview spans, that I marked in an orange rectangle, children of `viewAppearing`, as they are always executed during `viewAppearing`?

layoutSubView called after viewWillAppear are probably related to each other, but is not guaranteed, there is other situations where layoutSubView are called.

File: Sources/Sentry/SentryUIViewControllerPerformanceTracker.m [hunk 1]
```diff
@@ -15,6 +15,9 @@
 static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID
     = @"SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID";
 
+static NSString *const SENTRY_UI_PERFORMANCE_TRACKER_VIEWAPPEARING_SPAN_ID
+    = @"SENTRY_UI_PERFORMANCE_TRACKER_LAYOUTSUBVIEW_SPAN_ID";
```
Comment (philipphofmann) [inline Sources/Sentry/SentryUIViewControllerPerformanceTracker.m:19]: `h`: I think this should be. I wonder why the tests are not failing. Can we write a test that fails if this is messed up?
```suggestion
    = @"SENTRY_UI_PERFORMANCE_TRACKER_VIEWAPPEARING_SPAN_ID";
```

File: Sources/Sentry/SentryUIViewControllerPerformanceTracker.m [hunk 2]
```diff
@@ -65,18 +68,54 @@ - (void)viewControllerViewDidLoad:(UIViewController *)controller
...
 
-    if (spanId != nil) {
+    if (spanId == nil || ![self.tracker isSpanAlive:spanId]) {
+        // We are no longer tracking this UIViewController, just call the base method.
+        callback();
+    } else {
+        SentrySpanId *viewAppearingId = objc_getAssociatedObject(
+            controller, &SENTRY_UI_PERFORMANCE_TRACKER_VIEWAPPEARING_SPAN_ID);
+        if (viewAppearingId != nil) {
+            [self.tracker popActiveSpan]; // pop viewAppearingSpan pushed at viewWillAppear
```
Comment (philipphofmann) [inline Sources/Sentry/SentryUIViewControllerPerformanceTracker.m:107]: Useful comment thanks 🙏

**Review by philipphofmann — COMMENTED** [review]
Just one issue. Good job.

File: Sources/Sentry/SentryUIViewControllerPerformanceTracker.m (hunk 1 above)
Comment (brustolin) [reply Sources/Sentry/SentryUIViewControllerPerformanceTracker.m:19]: Good point.

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**